In [24]:
import pandas as pd 
import matplotlib as plt
import numpy as np
import seaborn as sns

# Importing Dataset and overview

To import F1 Drivers Dataset

In [19]:
df = pd.read_csv('F1DriversDataset.csv')


In [20]:
df

Driver     Nationality                         Seasons  \
0          Carlo Abate           Italy                    [1962, 1963]   
1     George Abecassis  United Kingdom                    [1951, 1952]   
2        Kenny Acheson  United Kingdom                    [1983, 1985]   
3    Andrea de Adamich           Italy  [1968, 1970, 1971, 1972, 1973]   
4       Philippe Adams         Belgium                          [1994]   
..                 ...             ...                             ...   
863      Emilio Zapico           Spain                          [1976]   
864        Zhou Guanyu           China                          [2022]   
865      Ricardo Zonta          Brazil  [1999, 2000, 2001, 2004, 2005]   
866        Renzo Zorzi           Italy              [1975, 1976, 1977]   
867     Ricardo Zunino       Argentina              [1979, 1980, 1981]   

     Championships  Race_Entries  Race_Starts  Pole_Positions  Race_Wins  \
0              0.0           3.0          0.0             0.0        0.0   
1              0.0           2.0          2.0             0.0        0.0   
2              0.0          10.0          3.0             0.0        0.0   
3              0.0          36.0         30.0             0.0        0.0   
4              0.0           2.0          2.0             0.0        0.0   
..             ...           ...          ...             ...        ...   
863            0.0           1.0          0.0             0.0        0.0   
864            0.0          23.0         23.0             0.0        0.0   
865            0.0          37.0         36.0             0.0        0.0   
866            0.0           7.0          7.0             0.0        0.0   
867            0.0          11.0         10.0             0.0        0.0   

     Podiums  Fastest_Laps  ...  Championship Years  Decade Pole_Rate  \
0        0.0           0.0  ...                 NaN    1960       0.0   
1        0.0           0.0  ...                 NaN    1950       0.0   
2        0.0           0.0  ...                 NaN    1980       0.0   
3        0.0           0.0  ...                 NaN    1970       0.0   
4        0.0           0.0  ...                 NaN    1990       0.0   
..       ...           ...  ...                 ...     ...       ...   
863      0.0           0.0  ...                 NaN    1980       0.0   
864      0.0           2.0  ...                 NaN    2020       0.0   
865      0.0           0.0  ...                 NaN    2000       0.0   
866      0.0           0.0  ...                 NaN    1980       0.0   
867      0.0           0.0  ...                 NaN    1980       0.0   

     Start_Rate  Win_Rate  Podium_Rate  FastLap_Rate  Points_Per_Entry  \
0      0.000000       0.0          0.0      0.000000          0.000000   
1      1.000000       0.0          0.0      0.000000          0.000000   
2      0.300000       0.0          0.0      0.000000          0.000000   
3      0.833333       0.0          0.0      0.000000          0.166667   
4      1.000000       0.0          0.0      0.000000          0.000000   
..          ...       ...          ...           ...               ...   
863    0.000000       0.0          0.0      0.000000          0.000000   
864    1.000000       0.0          0.0      0.086957          0.260870   
865    0.972973       0.0          0.0      0.000000          0.081081   
866    1.000000       0.0          0.0      0.000000          0.142857   
867    0.909091       0.0          0.0      0.000000          0.000000   

     Years_Active  Champion  
0               2     False  
1               2     False  
2               2     False  
3               5     False  
4               1     False  
..            ...       ...  
863             1     False  
864             1     False  
865             5     False  
866             3     False  
867             3     False  

[868 rows x 22 columns]

To show top 5 rows to get an idea of what dataset looks like, df.head(2) will just give top 2 rows

In [15]:
df.head()

Driver     Nationality                         Seasons  \
0        Carlo Abate           Italy                    [1962, 1963]   
1   George Abecassis  United Kingdom                    [1951, 1952]   
2      Kenny Acheson  United Kingdom                    [1983, 1985]   
3  Andrea de Adamich           Italy  [1968, 1970, 1971, 1972, 1973]   
4     Philippe Adams         Belgium                          [1994]   

   Championships  Race_Entries  Race_Starts  Pole_Positions  Race_Wins  \
0            0.0           3.0          0.0             0.0        0.0   
1            0.0           2.0          2.0             0.0        0.0   
2            0.0          10.0          3.0             0.0        0.0   
3            0.0          36.0         30.0             0.0        0.0   
4            0.0           2.0          2.0             0.0        0.0   

   Podiums  Fastest_Laps  ...  Championship Years  Decade Pole_Rate  \
0      0.0           0.0  ...                 NaN    1960       0.0   
1      0.0           0.0  ...                 NaN    1950       0.0   
2      0.0           0.0  ...                 NaN    1980       0.0   
3      0.0           0.0  ...                 NaN    1970       0.0   
4      0.0           0.0  ...                 NaN    1990       0.0   

   Start_Rate  Win_Rate  Podium_Rate  FastLap_Rate  Points_Per_Entry  \
0    0.000000       0.0          0.0           0.0          0.000000   
1    1.000000       0.0          0.0           0.0          0.000000   
2    0.300000       0.0          0.0           0.0          0.000000   
3    0.833333       0.0          0.0           0.0          0.166667   
4    1.000000       0.0          0.0           0.0          0.000000   

   Years_Active  Champion  
0             2     False  
1             2     False  
2             2     False  
3             5     False  
4             1     False  

[5 rows x 22 columns]

To get last 5 rows of info

In [21]:
df.tail()

Driver Nationality                         Seasons  \
863   Emilio Zapico       Spain                          [1976]   
864     Zhou Guanyu       China                          [2022]   
865   Ricardo Zonta      Brazil  [1999, 2000, 2001, 2004, 2005]   
866     Renzo Zorzi       Italy              [1975, 1976, 1977]   
867  Ricardo Zunino   Argentina              [1979, 1980, 1981]   

     Championships  Race_Entries  Race_Starts  Pole_Positions  Race_Wins  \
863            0.0           1.0          0.0             0.0        0.0   
864            0.0          23.0         23.0             0.0        0.0   
865            0.0          37.0         36.0             0.0        0.0   
866            0.0           7.0          7.0             0.0        0.0   
867            0.0          11.0         10.0             0.0        0.0   

     Podiums  Fastest_Laps  ...  Championship Years  Decade Pole_Rate  \
863      0.0           0.0  ...                 NaN    1980       0.0   
864      0.0           2.0  ...                 NaN    2020       0.0   
865      0.0           0.0  ...                 NaN    2000       0.0   
866      0.0           0.0  ...                 NaN    1980       0.0   
867      0.0           0.0  ...                 NaN    1980       0.0   

     Start_Rate  Win_Rate  Podium_Rate  FastLap_Rate  Points_Per_Entry  \
863    0.000000       0.0          0.0      0.000000          0.000000   
864    1.000000       0.0          0.0      0.086957          0.260870   
865    0.972973       0.0          0.0      0.000000          0.081081   
866    1.000000       0.0          0.0      0.000000          0.142857   
867    0.909091       0.0          0.0      0.000000          0.000000   

     Years_Active  Champion  
863             1     False  
864             1     False  
865             5     False  
866             3     False  
867             3     False  

[5 rows x 22 columns]

To get overall number of Rows and Columns. Here we have 868 rows and 22 columns

In [16]:
df.shape

(868, 22)

To get more info eg: column name and amount, if object, float, int or bool

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 868 entries, 0 to 867
Data columns (total 22 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Driver              868 non-null    object 
 1   Nationality         868 non-null    object 
 2   Seasons             868 non-null    object 
 3   Championships       868 non-null    float64
 4   Race_Entries        868 non-null    float64
 5   Race_Starts         868 non-null    float64
 6   Pole_Positions      868 non-null    float64
 7   Race_Wins           868 non-null    float64
 8   Podiums             868 non-null    float64
 9   Fastest_Laps        868 non-null    float64
 10  Points              868 non-null    float64
 11  Active              868 non-null    bool   
 12  Championship Years  34 non-null     object 
 13  Decade              868 non-null    int64  
 14  Pole_Rate           868 non-null    float64
 15  Start_Rate          868 non-null    float64
 16  Win_Rate

To get a list of column headings  Total = 22  info inside [] is a list

In [25]:
df.columns

Index(['Driver', 'Nationality', 'Seasons', 'Championships', 'Race_Entries',
       'Race_Starts', 'Pole_Positions', 'Race_Wins', 'Podiums', 'Fastest_Laps',
       'Points', 'Active', 'Championship Years', 'Decade', 'Pole_Rate',
       'Start_Rate', 'Win_Rate', 'Podium_Rate', 'FastLap_Rate',
       'Points_Per_Entry', 'Years_Active', 'Champion'],
      dtype='object')

df.describe will give some info like count, average and standard deviation

In [26]:
df.describe()

Championships  Race_Entries  Race_Starts  Pole_Positions   Race_Wins  \
count     868.000000    868.000000   868.000000      868.000000  868.000000   
mean        0.084101     29.917051    27.694700        1.244240    1.247696   
std         0.524883     53.780150    52.876476        6.347512    6.491921   
min         0.000000      1.000000     0.000000        0.000000    0.000000   
25%         0.000000      2.000000     1.000000        0.000000    0.000000   
50%         0.000000      7.000000     5.000000        0.000000    0.000000   
75%         0.000000     29.250000    26.000000        0.000000    0.000000   
max         7.000000    359.000000   356.000000      103.000000  103.000000   

          Podiums  Fastest_Laps       Points       Decade   Pole_Rate  \
count  868.000000    868.000000   868.000000   868.000000  868.000000   
mean     3.756912      1.261521    55.849459  1972.223502    0.011471   
std     14.432826      5.413644   265.968614    20.201084    0.046544   
min      0.000000      0.000000     0.000000  1950.000000    0.000000   
25%      0.000000      0.000000     0.000000  1960.000000    0.000000   
50%      0.000000      0.000000     0.000000  1970.000000    0.000000   
75%      0.000000      0.000000     8.000000  1982.500000    0.000000   
max    191.000000     77.000000  4415.500000  2020.000000    0.557692   

       Start_Rate    Win_Rate  Podium_Rate  FastLap_Rate  Points_Per_Entry  \
count  868.000000  868.000000   868.000000    868.000000        868.000000   
mean     0.779824    0.011050     0.041389      0.011894          0.479158   
std      0.324303    0.043579     0.106457      0.043514          1.213826   
min      0.000000    0.000000     0.000000      0.000000          0.000000   
25%      0.666667    0.000000     0.000000      0.000000          0.000000   
50%      0.962251    0.000000     0.000000      0.000000          0.000000   
75%      1.000000    0.000000     0.000000      0.000000          0.382517   
max      1.000000    0.461538     1.000000      0.500000         14.197749   

       Years_Active  
count    868.000000  
mean       3.664747  
std        3.501761  
min        1.000000  
25%        1.000000  
50%        2.000000  
75%        5.000000  
max       19.000000